coderd/wsbuilder/wsbuilder.go

```diff
@@ -632,6 +651,107 @@ func (b *Builder) getLastBuildJob() (*database.ProvisionerJob, error) {
 	return b.lastBuildJob, nil
 }
 
+func (b *Builder) getProvisionerTags() (map[string]string, error) {
+	// Step 1: Fetch required data
+	workspaceTags, err := b.getTemplateVersionWorkspaceTags()
+	if err != nil {
+		return nil, BuildError{http.StatusInternalServerError, "failed to fetch template version workspace tags", err}
+	}
+	parameterNames, parameterValues, err := b.getParameters()
+	if err != nil {
+		return nil, err // already wrapped BuildError
+	}
+	templateVersionJob, err := b.getTemplateVersionJob()
+	if err != nil {
+		return nil, BuildError{http.StatusInternalServerError, "failed to fetch template version job", err}
+	}
+	annotationTags := provisionersdk.MutateTags(b.workspace.OwnerID, templateVersionJob.Tags)
+
+	// Step 2: Evaluate provisioner tags
+	tags := map[string]string{}
+	for name, value := range annotationTags {
+		tags[name] = value
+	}
```

Comment:
Nit: Would make more sense to me for these to be first, and the rest (above/below) grouped together.